### ISSUE TITLE
Bootstrap carousel not working with lazyload

### ISSUE THREAD

**Issue by hiddenman** [verlok/vanilla-lazyload#451]
Hello,
Vanilla  lazload 15.1.1

I've got the same problem as describe there (bootstrap modal):
https://github.com/verlok/lazyload/issues/50

The solution for bootstrap modal works fine, because there is an event "shown" for the modal.
However, the carousel has only slide and slid event (https://getbootstrap.com/docs/4.0/components/carousel/#events ).  So it's possible to add a new instance of the lazyload only after showing at least one slide like this:

```js
    var carouselLazyLoad;
    $('#carousel').on('slide.bs.carousel', function () {
        carouselLazyLoad = new LazyLoad({
            elements_selector: ".lazy",
            container: this
        });
      });
```
Here is the simple solution: https://stackoverflow.com/questions/12697216/bootstrap-carousel-lazy-load
So it has one slide with real src, so then carousel shown it and the event "slide.bs.carousel" is fired.

Haven't found anything useful without showing first slide. Tried on load/change/ready and so on.

May be you've got an idea how to implement this easier?

Thank you.

**Comment by hiddenman** [comment]
Looks like i was wrong and even showing first slide doesn't help. Realized that this code from our developer breaks the lazyload working with carousels. Code is important for something on the site, not sure how to make lazyload working with it:

```
$('#carouse.carousel .carousel-item').each(function(){
        let next = $(this).next();
        if (!next.length) {
            next = $(this).siblings(':first');
        }
        next.children(':first-child').clone().appendTo($(this));

        for (let i=0;i<2;i++) {
            next=next.next();
            if (!next.length) {
                next = $(this).siblings(':first');
            }

            next.children(':first-child').clone().appendTo($(this));
        }
    });
```

**Comment by verlok** [comment]
Hey @hiddenman 
sorry for the late reply.

The code you've posted in your first message creates a new instance of LazyLoad on each slider, but you're specifying the scrolling container (`container`) to be the current carousel, whereas a carousel is not a scrolling container (with scrolling meaning the real browser scroll), so... you would probably just need to create a single instance of LazyLoad on your page.

```js
    // At page level...
    var carouselLazyLoad = new LazyLoad({
        elements_selector: ".lazy",
        // No container here! 👀
    });
```

The other code block you've posted clones the elements in the slider using Javascript, so you need to `update()` the `carouselLazyLoad` instance at that point.

```js
    $('#carouse.carousel .carousel-item').each(function(){
        let next = $(this).next();
        if (!next.length) {
            next = $(this).siblings(':first');
        }
        next.children(':first-child').clone().appendTo($(this));

        for (let i=0;i<2;i++) {
            next=next.next();
            if (!next.length) {
                next = $(this).siblings(':first');
            }

            next.children(':first-child').clone().appendTo($(this));
        }

        // The following line tells LazyLoad that new elements have been added to the page 👀
        carouselLazyLoad.update(); 
    });
```

**Comment by hiddenman** [comment]
Hi,

Sorry, missed you  answer. Thank you very much for your help! I will try this and let you know.

**Comment by epignosis567** [comment]
Hi, this solution did not work for me. I am also looking to lazy load / defer images in carousel inside modal in Boostrap 4.5
I like this because it’s vanilla js (although I’m not totally comfortable in jquery), but more importantly because it wouldn’t require me to drastically rewrite my code, the page is quite large and I’m hoping to be able to simply add a class to the images. 

Also, your code as written above has:
$('#carouse.carousel
should it be carousel.carousel ?

But regardless, this won’t work for me. Any advice (for a newbie)? I’ve uploaded my files to github: https://github.com/epignosis567/mygithub/tree/main/lazyLoadBootstrap

**Comment by hiddenman** [comment]
> Hi, this solution did not work for me. I am also looking to lazy load / defer images in carousel inside modal in Boostrap 4.5
> I like this because it’s vanilla js (although I’m not totally comfortable in jquery), but more importantly because it wouldn’t require me to drastically rewrite my code, the page is quite large and I’m hoping to be able to simply add a class to the images.
> 
> Also, your code as written above has:
> $('#carouse.carousel
> should it be carousel.carousel ?
> 
> But regardless, this won’t work for me. Any advice (for a newbie)? I’ve uploaded my files to github: https://github.com/epignosis567/mygithub/tree/main/lazyLoadBootstrap

Hi,

You should have replaced "src" with "data-src" for images, otherwise your browser will load them despite any JS code :)

**Comment by epignosis567** [comment]
Thanks! But it still doesn't work, now nothing loads in the carousel. Could you provide an example? I've been working on this for hours.

**Comment by hiddenman** [comment]
> Thanks! But it still doesn't work, now nothing loads in the carousel. Could you provide an example? I've been working on this for hours. Sorry the github file wasn't up to date I just uploaded it now.

Try to put this before the carousel launch:

```
   $('#myModal')
        .bind('shown.bs.modal', function () {
            modalLazyLoad = new LazyLoad({
                elements_selector: ".lazy",
                container: this
            });
        })
        .bind('hidden.bs.modal', function () {
            modalLazyLoad.destroy();
        });
```

**Comment by epignosis567** [comment]
Thanks! What does that do?

**Comment by hiddenman** [comment]
> Thanks! @hiddenman What does that do?
> Also I have a bunch of carousels on the full page, should I just put this between script tags in the header?
> 
> EDIT: I'm re-editing this comment for like the 100th time, I'm just going to open a new issue soon. If you could explain what the code you pasted above does, and where best to put it I would be very much appreciative.

The code adds new instance of the lazyload to a new container. Can't you see this?
